### PR TITLE
scheduler: avoid to updateCounts frequently

### LINKED ISSUE #7837: Lock contention problem when creating too many operators
## Enhancement Task

Consider that we create lots of scatter region operators, some of them are created and the others might be finished and removed. But they all need to lock the operator map. So there might be a lock contention problem due to the count update being O(n^2). See

img_v3_028a_03a9fd45-cc7c-49f3-9e7b-df3be43efc3g

img_v3_0289_5384bb66-ecff-4617-8ae2-694b0884c46g
img_v3_0289_09e0e749-4179-4ad6-ac6a-3704e8c6dbfg
img_v3_0289_9a5c36d9-20ba-4160-b6f5-0b1ac384f1bg

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: Close #7837

### What is changed and how does it work?

```commit-message
Use O(1) method to update `counts`.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

before PR, the speed of create operator decreased.
image

after PR
image

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

File: pkg/schedule/operator/operator_controller.go [hunk 1]
```diff
@@ -602,7 +602,7 @@ func (oc *Controller) removeOperatorLocked(op *Operator) bool {
 	regionID := op.RegionID()
 	if cur := oc.operators[regionID]; cur == op {
 		delete(oc.operators, regionID)
-		oc.updateCounts(oc.operators)
+		oc.counts[op.SchedulerKind()]--
```
Comment (HuSharp) [inline pkg/schedule/operator/operator_controller.go:605]: Why not removed `updateCounts` from `removeOperatorsLocked` and remove function `updateCounts` directly?

Comment (Leavrth) [reply pkg/schedule/operator/operator_controller.go:605]: done

**Review by HuSharp — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]

**Pull request merged**